bridge/engine/EngineFunction.h

```diff
@@ -0,0 +1,254 @@
+#include <cassert>
+#include <tuple>
+
+namespace bridge
+{
+
+class EngineFunction;
+
+namespace detail
+{
+
+constexpr size_t calculateStorageSize()
+{
+    constexpr size_t minSize = 128;
+    // EngineFunction will have EngineFunctionStorage + a pointer
```

Comment:
I am very much against these long explanations in between the lines of code. If really necessary, put them above the method and try to condense it. I am sure we can explain the same thing in half the number of words.